### PR TITLE
TS-fix-copy-and-select-groups-in-Semester-page

### PR DESCRIPTION
**Pull request by ViktoriiaKosovan**

### CODE REVIEW

**Review by Yashchemskyi-Oleksandr — APPROVED** [review]

**Review by HiroFive — APPROVED** [review]

File: frontend/src/components/RoomsPage/RoomForm/RoomForm.js [hunk 1]
```diff
@@ -67,11 +68,11 @@ const RoomForm = (props) => {
                     label={t(TYPE_LABEL)}
                     validate={[required]}
                 >
-                    <option className="room-select-item" />
+                    <MenuItem value="" className="hidden" disabled />
```
Comment (Akr1s) [inline frontend/src/components/RoomsPage/RoomForm/RoomForm.js:71]: Why do we need this strange item?

File: frontend/src/sagas/semesters.js [hunk 2]
```diff
@@ -160,10 +160,17 @@ export function* handleSemesterFormSubmit({ values }) {
             (semesterItem) =>
                 semesterItem.currentSemester === true && semesterItem.id !== values.id,
         );
+        const oldDefaultSemester = state.semesters.semesters.find(
+            (semesterItem) =>
+                semesterItem.defaultSemester === true && semesterItem.id !== values.id,
```
Comment (Akr1s) [inline frontend/src/sagas/semesters.js]: we don't need to compare to `true`

File: frontend/src/sagas/semesters.js [hunk 3]
```diff
@@ -160,10 +160,17 @@ export function* handleSemesterFormSubmit({ values }) {
...
+        );
         if (values.currentSemester && oldCurrentSemester) {
             oldCurrentSemester.currentSemester = false;
-            const { data } = yield call(axiosCall, SEMESTERS_URL, PUT, oldCurrentSemester);
-            yield put(updateSemesterSuccess(data));
+            yield put(updateSemesterSuccess(oldCurrentSemester));
+        }
+        if (values.defaultSemester && oldDefaultSemester) {
+            oldDefaultSemester.defaultSemester = false;
```
Comment (Akr1s) [inline frontend/src/sagas/semesters.js:174]: Data mutation

File: frontend/src/sagas/semesters.js [hunk 4]
```diff
@@ -173,10 +180,18 @@ export function* handleSemesterFormSubmit({ values }) {
 
 export function* setDefaultSemesterById({ semesterId }) {
     try {
+        const state = yield select();
+        const oldDefaultSemester = state.semesters.semesters.find(
+            (semesterItem) =>
+                semesterItem.defaultSemester === true && semesterItem.id !== semesterId,
+        );
+        if (oldDefaultSemester) {
+            oldDefaultSemester.defaultSemester = false;
```
Comment (Akr1s) [inline frontend/src/sagas/semesters.js]: As above

File: frontend/src/components/RoomsPage/RoomForm/RoomForm.js (hunk 1 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/RoomsPage/RoomForm/RoomForm.js:71]: it's from material. I use it for style options in select and add effect of hover.

File: frontend/src/sagas/semesters.js (hunk 2 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js]: fixed

File: frontend/src/sagas/semesters.js (hunk 3 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js:172]: I need it for set old defaultSemester to false and set new default semester to true without doing call to backend

File: frontend/src/sagas/semesters.js (hunk 4 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js]: as above

File: frontend/src/sagas/semesters.js (hunk 3 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js:174]: fixed

File: frontend/src/sagas/semesters.js (hunk 4 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js]: fixed

File: frontend/src/components/Semester/SemesterSidebar.js [hunk 5]
```diff
@@ -56,7 +56,7 @@ const SemesterSidebar = (props) => {
     };
 
     return (
-        <aside className="search-list__panel">
+        <aside className="semester-aside__panel">
```
Comment (Niknikson) [inline frontend/src/components/Semester/SemesterSidebar.js]: semester-aside__panel  -> semester-aside-panel

File: frontend/src/reducers/rooms.js [hunk 6]
```diff
@@ -13,7 +13,7 @@ const reducer = (state = initialState, action) => {
         case actionTypes.ADD_ROOM:
             return {
                 ...state,
-                rooms: [...state.rooms, action.room].sort((a, b) => a.name.localeCompare(b.name)),
+                rooms: [...state.rooms, action.room],
```
Comment (Niknikson) [inline frontend/src/reducers/rooms.js]: Are you sure added room will be add to the end of array ?

**Review by Niknikson — APPROVED** [review]

File: frontend/src/reducers/rooms.js (hunk 6 above)
Comment (ViktoriiaKosovan) [reply frontend/src/reducers/rooms.js]: fixed

File: frontend/src/components/Semester/SemesterSidebar.js (hunk 5 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/Semester/SemesterSidebar.js]: fixed

**Review by HiroFive — APPROVED** [review]

**Review by andrii-kuzmichov — APPROVED** [review]

**Pull request merged**